proxy/src/telemetry/metrics/histogram.rs

```diff
@@ -88,6 +88,113 @@ impl<V: Into<u64>> Histogram<V> {
         self.buckets[idx].incr();
         self.sum += value;
     }
+
+    // ===== Test-only methods to help with assertions about histograms. =====
+
+    /// Assert the bucket containing `le` has a count of at least `at_least`.
+    #[cfg(test)]
+    pub fn assert_bucket_at_least(&self, le: u64, at_least: u64) {
+        for (&bucket, &count) in self {
+            let is_le = if let Bucket::Le(ceiling) = bucket {
```

Comment:
Can you factor out the repeated `is_le` logic?  Tests would be more readable with `if le < bucket { ... }` or something that effect.